.github/workflows/build.yml

```diff
@@ -33,7 +33,7 @@ jobs:
         run: |
           ver=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout -f rta)
           if [[ $ver == *"SNAPSHOT"* ]]; then
-              mvn -B -ntp deploy -f rta
+              mvn -B -ntp -Dmaven.test.skip=true deploy -f rta
```

Comment:
I was checking a [previous run](https://github.com/gluonhq/rich-text-area/actions/runs/8967621767/job/24625481389) and it seems that the "Build project" phase don't run the tests. These are actually ran in "Publish Snapshots" step.

Considering this, I am of the opinion that we should remove the previous all together and make the `build.yml` similar to `release.yml` where both tests and deployment are done using the same step.

Reply:
It does run them: https://github.com/gluonhq/rich-text-area/actions/runs/8967621767/job/24625481389#step:4:34

Reply:
Ah, my bad. However, they [run again](https://github.com/gluonhq/rich-text-area/actions/runs/8967621767/job/24625481389#step:5:54) in the next step.

I would prefer to remove the previous step to consistency across workflows. Thoughts?

Reply:
We can't remove `Build project`, as that is used for PRs. `Publish Snapshots` only applies when committed to main.

Reply:
Ah, now I see why it was divided into 2 steps to begin with. Thank you for reminding me :)